src/main/java/com/internet/shop/dao/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,190 @@
+                    + " was not created updated", e);
+        }
+        deleteProductsInShoppingCart(shoppingCart.getId());
+        return addProductsIntoSHoppingCart(shoppingCart);
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "UPDATE shoping_cart "
+                    + "SET deleted = TRUE WHERE id_shoping_cart = ?";
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() != 0;
+        } catch (SQLException e) {
+            throw new DataBaseProcessingException("Shopping cart is was not deleted", e);
+        }
+    }
+
+    private List<Product> getCartProducts(Long id) {
+        List<Product> products = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "SELECT * FROM products JOIN shoping_cart_products "
+                    + "ON products.product_id = shoping_cart_products.id_product "
+                    + "WHERE shoping_cart_products.id_cart=?";
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                Product product = new Product(
+                        resultSet.getString("name"),
+                        resultSet.getDouble("price"));
+                product.setId(resultSet.getLong("product_id"));
+                products.add(product);
+            }
+        } catch (SQLException e) {
+            throw new DataBaseProcessingException("Shoping cart "
+                    + id + " have problem which product list", e);
+        }
+        return products;
+    }
+
+    private boolean deleteProductsInShoppingCart(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "DELETE FROM shoping_cart_products WHERE id_cart = ?";
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, id);
+            return preparedStatement.executeUpdate() != 0;
+        } catch (SQLException e) {
+            throw new DataBaseProcessingException("Shopping product in cart "
+                    + id + " was not deleted", e);
+        }
+    }
+
+    private ShoppingCart addProductsIntoSHoppingCart(ShoppingCart shoppingCart) {
+        String query = "INSERT INTO shoping_cart_products (id_cart, id_product) "
+                + "VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+
+            for (Product product : shoppingCart.getProducts()) {
```

Comment:
```suggestion
            PreparedStatement statement = connection.prepareStatement(query);
            for (Product product : shoppingCart.getProducts()) {
```

Reply:
removed empty lin in ShoppingCartDaoJdbcImpl